lib/puppet/provider/package/pip.rb

```diff
@@ -36,11 +36,10 @@ def self.instances
   end
 
   def self.cmd
-    case Facter.value(:osfamily)
-      when "RedHat"
-        "pip-python"
-      else
-        "pip"
+    if Facter.value(:osfamily) == "RedHat" and Facter.value(:operatingsystemmajrelease).to_i < 7
+      "pip-python"
```

Comment:
This doesn't necessarily cover Fedora. A more robust solution would be to try to find pip (which), and if not found use pip-python.